### PR TITLE
enable schema tracking by default

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description
This PR enables schema tracking by default. It's mostly about changing default settings and then updating integration tests that start failing.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/vreplication/resharding_workflows_v2_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ const (
 	reverseKsWorkflow = sourceKs + "." + workflowName + "_reverse"
 	tablesToMove      = "customer"
 	defaultCellName   = "zone1"
-	readQuery         = "select * from customer"
+	readQuery         = "select cid from customer"
```
Comment (deepthi) [inline go/test/endtoend/vreplication/resharding_workflows_v2_test.go:42]: Why is this change required?

Comment (harshit-gangal) [reply go/test/endtoend/vreplication/resharding_workflows_v2_test.go:42]: the test expectation was not matching as the actual query getting executed is the expanded one with cid instead of `star`

This is what I can understand from Rohit's commit message.

**Review by systay — APPROVED** [review]
